src/app/listener.ts

```diff
@@ -42,6 +45,29 @@ export class Listener {
 
       this.container.loggerService.info(`Loaded ${this.container.jobService.size()} jobs...`);
 
+      const commands = Object.entries(this.container.pluginService.plugins)
+        .filter((entry) => {
+          const [, plugin] = entry;
+          return isSlashCommand(plugin);
+        })
+        .map((entry) => {
+          const [name, plugin] = entry;
+          const options = isSlashCommand(plugin)
+            ? (plugin as unknown as ISlashPlugin).parameters
+            : undefined;
+
+          return {
+            name: name,
+            description: plugin.description.substring(0, 99),
+            options,
+          };
+        });
+
+      // Register commands for all guilds.
+      for (const guild of this.container.clientService.guilds.cache.values()) {
+        guild.commands.set(commands);
```

Comment:
is this synchronous ? 

also do we have to do a pass of unregistering? otherwise a command that gets removed could be left dangling.

Reply:
Set should be a strict equals to, not add.
So any slash command that was not in the set will be removed

Reply:
I'm adding await just to be sure here, +1 on tanndlin's comment